utils/generator/README.md

```diff
@@ -0,0 +1,2638 @@
+```
+
+The constructor should include all set of `<param>` with the `"mandatory"` attribute is "true". JavaDoc should include all constructor parameters and the constructor should call all corresponding setters inside itself.
+
+Template:
+```java
+    /**
+     * Constructs a new [name] object
+     *
+     * @param [param_name]
+     */
+    public [name](@NonNull [param_type|List<[param_type]>] [param_name]) {
+        this();
+        [setter_name]([param_name]);
+    }
+```
+Where:
+* `[name]` is the value from the `"name"` attribute of `<struct>`.
+* `[param_name]` is `"name"` attribute of the `<param>`.
+* `[param_type]` is `"type"` attribute of the `<param>`, `[List<[param_type]>]` applied if `"array"` attribute of `<param>` is "true".
+* `[setter_name]` is the name of the corresponding setter method
+
+For each `<param>` the getter and setter methods should be defined in the class:
+
+1. The name of the setter/getter is the `PascalCase` formatted value of the `"name"` attribute with the `get` prefix for the getter, for the setter the prefix should be `set`.
+1. Uses of the "sync" prefix shall be replaced with "sdl" (where it would not break functionality). E.g. `SyncMsgVersion -> SdlMsgVersion`. This applies to member variables and their accessors. The key used when creating the RPC message JSON should match that of the RPC Spec.
+1. The setter method:
+    * Accepts the single parameter with the type defined in the `"type"` attribute and the name defined in the `"name"` attribute of the `<param>`;
+    * The parameter should be decorated by `@NonNull` annotation if the `"mandatory"` attribute of the `<param>` is "true";
+    * Should call the `setValue` method, where the first parameter is the value of the corresponding static field described above, the second is the value passed into setter;
+1. The getter method:
+    * If `"type"` attribute of the `<param>` has the one of `Boolean`, `Integer` or `String`
+        * the getter should call and return the corresponding `getBoolean`, `getInteger` or `getString` method, the parameter of that method is the value of the corresponding static field described above;
+    * If `"type"` attribute of the `<param>` is `Float`:
+        * the getter should call the `getValue` method, the parameter of that method is the value of the corresponding static field described above;
+        * the getter should return `SdlDataTypeConverter.objectToFloat(object)` where the `object` is the value previously received from `getValue`;
+    * If the `<param>` has the `"type"` attribute value as the one of `<enum>` or `<struct>` name:
+        * The getter should call and return the result of the `getObject` method, where the first parameter is the corresponding Struct or Enum class, the second is the value of the corresponding static field described above;
+
+Setter template:
+```java
+    /**
+     * Sets the [name].
+     * [description]
+     *
+     * @param [name]
```

Comment:
Hello @bilal-alsharifi 
parameter description moved to getter/setter/constructor javadoc as you described

Reply:
Thanks @ksologubov for making the change

Reply:
@ksologubov  would it be possible to make the param description starts on the same line as the param? that applies to the @retrun as well.

Reply:
also, if we can move the `@since comments` from the `public static final String` keys to the getter/setters as well. Because developers will look at the docs for the methods not the keys

Reply:
So the expected result for `iconResumed` for example will be something like this :

```
public static final String KEY_ICON_RESUMED = "iconResumed";

/**
 * Sets the iconResumed.
 *
 * @param iconResumed Existence of apps icon at system. If true, apps icon was resumed at system.
 * If false, apps icon is not resumed at system
 *
 * @since SmartDeviceLink 5.0.0
 */
public void setIconResumed(Boolean iconResumed) {
    setParameters(KEY_ICON_RESUMED, iconResumed);
}

/**
 * Gets the iconResumed.
 *
 * @return Boolean Existence of apps icon at system. If true, apps icon was resumed at system.
 * If false, apps icon is not resumed at system
 *
 * @since SmartDeviceLink 5.0.0
 */
public Boolean getIconResumed() {
    return getBoolean(KEY_ICON_RESUMED);
}
```

Reply:
Thanks for example Bilal, I'm going to update this

Reply:
@bilal-alsharifi Generator was corrected per your comments